jetty-util/src/main/java/org/eclipse/jetty/util/security/Credential.java

```diff
@@ -187,7 +187,14 @@ public static String crypt(String user, String pw)
         MD5(String digest)
         {
             digest = digest.startsWith(__TYPE) ? digest.substring(__TYPE.length()) : digest;
-            _digest = TypeUtil.parseBytes(digest, 16);
+            // intentionally not using TypeUtil.parseBytes as using TypeUtil introduces
+            // a logging requirement that is not supported by Command Line Password util
+            byte[] digestBytes = new byte[digest.length() / 2];
+            for (int i = 0; i < digest.length(); i += 2)
+            {
+                digestBytes[i / 2] = (byte)TypeUtil.parseInt(digest, i, 2, 16);
```

Comment:
Seems strange to have a comment saying that we can't use `TypeUtil`, but then we use it.
`Integer.parseInt(digest.substring(i, i+2), 16)` does the same, so I'd get rid of `TypeUtil` usage here.
I would also drop the comment, as referencing a class that is not used makes the comment weird imho.

Reply:
I want to break apart TypeUtil.

The String / Hex methods should exist in StringUtil (which has no logging).

The TypeUtil handling of types, and service loaders, and whatnot can stay.